File: Sorting/bubble_sort.py
#Bubble sort
def bubble_sort(arr):
    for i in range(len(arr)):
        for j in range(len(arr)-i-1):
            if arr[j] > arr[j+1]:
                temp = arr[j]
                arr[j] = arr[j+1]
                arr[j+1] = temp
    return arr

#print(bubble_sort([10,5,6,1,7]))

def improved_bubble_sort(arr):
    swapped = True
    while swapped:
        swapped = False
        for i in range(len(arr)-1):
            if arr[i] > arr[i+1]:
                temp = arr[i]
                arr[i] = arr[i+1]
                arr[i+1] = temp
                swapped = True
    return arr

print(improved_bubble_sort([5,4,3,2,1]))

File: Sorting/bubble_sort.java
import java.lang.*;
public class Sort
{
    public static void bubble_sort(int[] arr)
    {
        int temp;
        boolean swapped;
        do
        {
            swapped=false;
            for(int i = 0; i < arr.length-1; i++)
            {
                if(arr[i] > arr[i+1])
                {
                    temp = arr[i];
                    arr[i] = arr[i+1];
                    arr[i+1] = temp;
                    swapped = true;
                }
            }
        }while(swapped);
        for(int j=0; j< arr.length; j++){
            System.out.println(arr[j]);
        }
    }

    public static void main(String[] args)
    {
        int[] arr = {4,5,1,3,2, 1};
        bubble_sort(arr);
    }
}